src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -884,23 +884,36 @@ void IslandWindow::_RestoreFullscreenPosition(const RECT rcWork)
                rcWork.left - _rcWorkBeforeFullscreen.left,
                rcWork.top - _rcWorkBeforeFullscreen.top);
 
+    const til::size ncSize{ GetTotalNonClientExclusiveSize(dpiWindow) };
+
+    RECT rcWorkAdjusted = rcWork;
+
+    // GH#10199 - adjust the size of the "work" rect by the size of our borders.
+    // We want to make sure the window is restored within the bounds of the
+    // monitor we're on, but it's totally fine if the invisible borders are
+    // outside the monitor.
+    rcWorkAdjusted.left -= ncSize.width<long>() / 2;
+    rcWorkAdjusted.right += ncSize.width<long>() / 2;
+    rcWorkAdjusted.top -= ncSize.height<long>() / 2;
+    rcWorkAdjusted.bottom += ncSize.height<long>() / 2;
+
     // Enforce that our position is entirely within the bounds of our work area.
     // Prefer the top-left be on-screen rather than bottom-right (right before left, bottom before top).
-    if (rcRestore.right > rcWork.right)
+    if (rcRestore.right > rcWorkAdjusted.right)
     {
-        OffsetRect(&rcRestore, rcWork.right - rcRestore.right, 0);
+        OffsetRect(&rcRestore, rcWorkAdjusted.right - rcRestore.right, 0);
     }
-    if (rcRestore.left < rcWork.left)
+    if (rcRestore.left < rcWorkAdjusted.left)
     {
-        OffsetRect(&rcRestore, rcWork.left - rcRestore.left, 0);
+        OffsetRect(&rcRestore, rcWorkAdjusted.left - rcRestore.left, 0);

```

Comment:
This fails to work properly if `rcRestore` covers a larger area than `rcWorkAdjusted` doesn't it?

Reply:
I don't think so... I think the way it already was, the whole point was clipping `rcRestore` inside of `rcWork`. So I think it only _ever_ did something when `rcWork` was smaller than `rcRestore`.

Now we're just making `rcWork` a little bigger, so we can keep the invisible borders off the monitor